tests/components/diagnostics/__init__.py

```diff
@@ -39,6 +43,34 @@ async def get_diagnostics_for_config_entry(
     return cast(JsonObjectType, data["data"])
 
 
+async def snapshot_get_diagnostics_for_config_entry(
+    hass: HomeAssistant,
+    hass_client: ClientSessionGenerator,
+    config_entry: ConfigEntry,
+    snapshot: SnapshotAssertion,
+) -> JsonObjectType:
+    """Snapshot get_diagnostics_for_config_entry."""
+    snapshot_exclude_functions: list[PropertyFilter] = [
+        props("created_at", "entry_id", "modified_at")  # we don't care about these
+    ]
+
+    if snapshot._exclude:
+        # if snaspshot has exclude functions, add them to the list
+        snapshot_exclude_functions.append(snapshot._exclude)
+
+    def prop_filter(*, prop: PropertyName, path: PropertyPath) -> bool:
+        """Filter out properties."""
+        return any(
+            exclude(prop=prop, path=path) for exclude in snapshot_exclude_functions
+        )
+
```

Comment:
Instead of doing this, why dont we update `syrupy.py` to handle `ConfigEntry` objects differently?

Reply:
It is already present in `syrupy.py` but will not work for diagnostics as the return type is a generic dict without any schema. Some integrations include the config entry under the `entry` key, others under the `config_entry` key,...

So the only solution I found is the above one